arduino/libraries/WiFi/src/WiFi.cpp

```diff
@@ -338,7 +339,13 @@ void WiFiClass::setDNS(/*IPAddress*/uint32_t dns_server1, /*IPAddress*/uint32_t
 
 void WiFiClass::hostname(const char* name)
 {
-  tcpip_adapter_set_hostname(_interface == ESP_IF_WIFI_AP ? TCPIP_ADAPTER_IF_AP : TCPIP_ADAPTER_IF_STA, name);
+  if (sizeof(name) <= HOSTNAME_MAX_LENGHT) {
```

Comment:
I believe here we should use `strlen(name)` instead of `sizeof`.

Reply:
done

Reply:
>done

I guess the changes are only in your local clone. It would be a good idea to push them so reviewers can see the current code for your PR.

Reply:
yes per, there is one review left actually, for this i don't have push the change.